src/java/org/apache/cassandra/streaming/StreamSession.java

```diff
@@ -830,4 +893,57 @@ public int getNumTransfers()
     {
         return transferredRangesPerKeyspace.size();
     }
+
+    @VisibleForTesting
+    public static class MessageStateSink
```

Comment:
Making the code easier to test is a good thing. We should just avoid having test related code within the production code.
I would have prefered to have a listener interface here and the implementation at the test level.

Reply:
that's what I tried initially. 

The jvm-dtest currently don't support fetching non-primitive objects from server jvm to test jvm, because of class loading..

One way is to put verification code in the implementation and let it execute on server jvm. I will update it.